app/database/operator/server_data_operator/transformers/thread.ts

```diff
@@ -100,18 +100,18 @@ export const transformThreadInTeamRecord = ({action, database, value}: Transform
         tableName: THREADS_IN_TEAM,
         value,
         fieldsMapper,
-    }) as Promise<ThreadInTeamModel>;
+    });
 };
 
-export const transformTeamThreadsSyncRecord = ({action, database, value}: TransformerArgs): Promise<TeamThreadsSyncModel> => {
-    const raw = value.raw as TeamThreadsSync;
-    const record = value.record as TeamThreadsSyncModel;
+export const transformTeamThreadsSyncRecord = ({action, database, value}: TransformerArgs<TeamThreadsSyncModel, TeamThreadsSync>): Promise<TeamThreadsSyncModel> => {
+    const raw = value.raw;
+    const record = value.record;
     const isCreateAction = action === OperationType.CREATE;
 
     const fieldsMapper = (teamThreadsSync: TeamThreadsSyncModel) => {
-        teamThreadsSync._raw.id = isCreateAction ? (raw?.id ?? teamThreadsSync.id) : record.id;
-        teamThreadsSync.earliest = raw.earliest || record?.earliest;
-        teamThreadsSync.latest = raw.latest || record?.latest;
+        teamThreadsSync._raw.id = isCreateAction ? (raw?.id ?? teamThreadsSync.id) : record?.id || '';
+        teamThreadsSync.earliest = raw.earliest || record?.earliest || 0;
+        teamThreadsSync.latest = raw.latest || record?.latest || 0;
```

Comment:
What happens if set to 0?

Reply:
I took another look at this. a 0 is not great, but should not happen. The logic is...
- Prefer whatever we just receive
- If we receive any 0, prefer whatever we have in the record
- In any other case, put a 0

But we should never be in the situation where what we receive and what we have is 0.